calour/training.py

```diff
@@ -89,6 +85,65 @@ def onehot_encode_features(exp: Experiment, fields, sparse=None, inplace=False):
     --------
     :class:`sklearn.preprocessing.OneHotEncoder`
     '''
+    # '''Add covariates from sample metadata to the data table as features for machine learning.
+
+    # This will convert the columns of categorical strings using one-hot encoding scheme and add them
+    # into the data table as new features.
+
+    # Examples
+    # --------
+    # >>> exp = Experiment(np.array([[1,2], [3, 4]]), sparse=False,
+    # ...                  sample_metadata=pd.DataFrame({'category': ['A', 'B'],
+    # ...                                                'ph': [6.6, 7.7]},
+    # ...                                               index=['s1', 's2']),
+    # ...                  feature_metadata=pd.DataFrame({'motile': ['y', 'n']}, index=['otu1', 'otu2']))
+    # >>> exp
+    # Experiment
+    # ----------
+    # data dimension: 2 samples, 2 features
+    # sample IDs: Index(['s1', 's2'], dtype='object')
+    # feature IDs: Index(['otu1', 'otu2'], dtype='object')
+
+    # Let's add the columns of `category` and `ph` as features into data table:
+
+    # >>> new = exp.onehot_encode_features(['category', 'ph'])
+    # >>> new
+    # Experiment
+    # ----------
+    # data dimension: 2 samples, 5 features
+    # sample IDs: Index(['s1', 's2'], dtype='object')
+    # feature IDs: Index(['category=A', 'category=B', 'ph', 'otu1', 'otu2'], dtype='object')
+    # >>> new.feature_metadata
+    #            motile
+    # category=A    NaN
+    # category=B    NaN
+    # ph            NaN
+    # otu1            y
+    # otu2            n
+    # >>> new.data
+    # array([[1. , 0. , 6.6, 1. , 2. ],
+    #        [0. , 1. , 7.7, 3. , 4. ]])
+
+    # Parameters
+    # ----------
+    # fields : list of str
+    #     the column names in the sample metadata. These columns will be
+    #     converted to one-hot numeric code and then concatenated to the
+    #     data table
+    # sparse : bool or ``None`` (optional)
+    #     use sparse or dense data matrix. When it is ``None``, it will follow
+    #     the same sparsity of the current data table in the :class:`.Experiment` object
+    # inplace : bool
+    #     change the :class:`.Experiment` object in place or return a copy of changed.
+
+    # Returns
+    # -------
+    # :class:`.Experiment`
+
+    # See Also
+    # --------
+    # :class:`sklearn.preprocessing.OneHotEncoder`
```

Comment:
understand you changed repr, but why other lines of docstring are commented out?

Reply:
uncommented and fixed to new __repr__ results